x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -1839,10 +1839,9 @@ public void testCombineOrderByThroughFilter() {
 
     /**
      * Expected
-     * TopN[[Order[first_name{f}#170,ASC,LAST]],1000[INTEGER]]
-     *  \_MvExpand[first_name{f}#170]
-     *    \_TopN[[Order[emp_no{f}#169,ASC,LAST]],1000[INTEGER]]
```

Comment:
Notice that here you are losing a performance aspect: `mv_expand` is applied on all documents. The idea with a sort (and the **default** pushed down `limit`) in front of `mv_expand` was to have fewer documents to expand from because the expansion is, theoretically, creating even more rows. Before this PR, each node was getting 1000 docs, mv_expanded them and a final sort performed. With this PR, each node is expanding everything, then sorting.